plug/pip_resolve.py

```diff
@@ -201,15 +201,18 @@ def create_dependencies_tree_by_req_file_path(requirements_file_path,
     required = get_requirements_list(requirements_file_path, dev_deps=dev_deps)
     installed = [p for p in dist_index]
     packages = []
+    missing_packages = []
     for r in required:
         if r.lower() not in installed:
-            msg = 'Required package missing: ' + r.lower()
-            if allow_missing:
-                sys.stderr.write(msg + "\n")
-            else:
-                sys.exit(msg)
+            missing_packages.append(r)
         else:
             packages.append(r)
+    if missing_packages:
+        msg = 'Required packages missing: ' + (', '.join(missing_packages))
```

Comment:
this changes the output (commas instead of newlines), but think it's ok - it was previously consumer only by `pip-deps` for logging if I'm not mistaken